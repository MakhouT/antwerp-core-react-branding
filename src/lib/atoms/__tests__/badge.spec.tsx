import * as React from "react";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Badge", () => {
    it("Renders default, string", () => {
        // Act
        const element = (
            <A.Badge
                value="value"
            />);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders default, number", () => {
        // Act
        const element = (
            <A.Badge
                value={10}
            />);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders badge with level danger", () => {
        // Arrang
        const level = A.Levels.Danger;

        // Act
        const element = (
            <A.Badge
                value={10}
                level={level}
            />);
        const tree = renderAndMatchSnapshot(element);

        // Assert
        expect(tree.props.className).toContain(`a-badge--${level}`);
        expect(tree).toMatchSnapshot();
    });

    it("Renders badge with level warning", () => {
        // Arrang
        const level = A.Levels.Warning;

        // Act
        const element = (
            <A.Badge
                value={10}
                level={level}
            />);
        const tree = renderAndMatchSnapshot(element);

        // Assert
        expect(tree.props.className).toContain(`a-badge--${level}`);
        expect(tree).toMatchSnapshot();
    });
});
