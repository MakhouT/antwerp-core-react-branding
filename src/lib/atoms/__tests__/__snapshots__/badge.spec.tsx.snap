// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Renders badge with level danger 1`] = `
<span
  className="a-badge a-badge--danger"
>
  10
</span>
`;

exports[`Badge Renders badge with level danger 2`] = `
<span
  className="a-badge a-badge--danger"
>
  10
</span>
`;

exports[`Badge Renders badge with level warning 1`] = `
<span
  className="a-badge a-badge--warning"
>
  10
</span>
`;

exports[`Badge Renders badge with level warning 2`] = `
<span
  className="a-badge a-badge--warning"
>
  10
</span>
`;

exports[`Badge Renders default, number 1`] = `
<span
  className="a-badge"
>
  10
</span>
`;

exports[`Badge Renders default, string 1`] = `
<span
  className="a-badge"
>
  value
</span>
`;
